### ISSUE TITLE
wca-prm agent crash after monitoring for 2 hours 

### ISSUE THREAD

**Issue by thuang6** [intel/workload-collocation-agent#150]
```
2019-06-06 02:41:03,648 DEBUG    {MainThread} [prm.allocator] prm allocate called...
2019-06-06 02:41:03,648 DEBUG    {MainThread} [prm.allocator] platform=Platform(sockets=2, cores=44, cpus=88, cpu_model='Intel(R) Xeon(R) CPU E5-2699 v4 @ 2.20GHz', cpus_usage={0: 4580486, 1: 2316501, 2: 4975647, 3: 1247623, 4: 4686022, 5: 2052125, 6: 3691205, 7: 2400870, 8: 2892963, 9: 1285943, 10: 2424433, 11: 1117333, 12: 3046447, 13: 1076388, 14: 2559407, 15: 1059773, 16: 3433984, 17: 1059321, 18: 3681235, 19: 1048567, 20: 2571104, 21: 1036494, 22: 2590433, 23: 1041747, 24: 4261469, 25: 1036596, 26: 3221641, 27: 1035310, 28: 3607573, 29: 1033610, 30: 4519563, 31: 1039658, 32: 2594587, 33: 1036916, 34: 2928363, 35: 1021574, 36: 4257606, 37: 1038865, 38: 2781840, 39: 1045555, 40: 2749580, 41: 1068567, 42: 2680857, 43: 1070989, 44: 3535391, 45: 955722, 46: 3041853, 47: 1989106, 48: 2669516, 49: 1606934, 50: 3916947, 51: 1452454, 52: 4326790, 53: 1059224, 54: 4541243, 55: 1007768, 56: 2494489, 57: 993072, 58: 3787067, 59: 984114, 60: 3204747, 61: 968221, 62: 4159426, 63: 972592, 64: 1823248, 65: 959419, 66: 1832040, 67: 955499, 68: 2305078, 69: 956293, 70: 1813937, 71: 952224, 72: 1824743, 73: 949903, 74: 1912272, 75: 942308, 76: 2187699, 77: 948239, 78: 1827496, 79: 935341, 80: 1851504, 81: 927092, 82: 1844873, 83: 927394, 84: 1872063, 85: 922491, 86: 1878746, 87: 920496}, total_memory_used=28357525504, timestamp=1559788863.6303248, rdt_information=RDTInformation(rdt_cache_monitoring_enabled=True, rdt_mb_monitoring_enabled=True, rdt_cache_control_enabled=True, rdt_mb_control_enabled=False, cbm_mask='fffff', min_cbm_bits='1', num_closids=16, mb_bandwidth_gran=None, mb_min_bandwidth=None))
2019-06-06 02:41:03,648 DEBUG    {MainThread} [prm.allocator] tasks_resources={'root-staging15-redis_rpc_perf-default--redis--6791-0-0-53b13ed6-c73a-4502-add5-4175ab23ed0b': {'mem': 2048.0, 'cpus': 1.0, 'disk': 2048.0}, 'root-staging15-redis_rpc_perf-default--redis--6789-0-0-90aa8f45-1aba-4024-8854-228bb553e084': {'mem': 2048.0, 'cpus': 1.0, 'disk': 2048.0}, 'root-staging15-twemcache_mutilate-small--twemcache--11211-0-0-c5253ea3-b401-4443-b792-af0f970f322f': {'mem': 51200.0, 'cpus': 2.0, 'disk': 51200.0}, 'root-staging15-cassandra_stress-default--cassandra--9142-0-0-2cf6f910-87c0-44aa-b5f0-230cbcea14e9': {'mem': 51200.0, 'cpus': 10.0, 'disk': 51200.0}, 'root-staging15-twemcache_mutilate-small--twemcache--11212-0-0-07cdf61b-6832-4639-ae97-a81b65260cd7': {'mem': 51200.0, 'cpus': 2.0, 'disk': 51200.0}, 'root-staging15-redis_rpc_perf-default--redis--6790-0-0-69f60671-18d4-488a-ad2e-740cd0620262': {'mem': 2048.0, 'cpus': 1.0, 'disk': 2048.0}, 'root-staging15-twemcache_mutilate-big--twemcache--11213-0-0-a9a7244a-ef48-4238-a310-d5313d1957ba': {'mem': 51200.0, 'cpus': 4.0, 'disk': 51200.0}}
2019-06-06 02:41:03,648 DEBUG    {MainThread} [prm.allocator] tasks_labels={'root-staging15-redis_rpc_perf-default--redis--6791-0-0-53b13ed6-c73a-4502-add5-4175ab23ed0b': {'aurora_tier': 'preemptible', 'job_name': 'redis_rpc_perf.default--redis--6791.0', 'workload_instance': 'redis_rpc_perf--default--15--6791', 'own_ip': '100.64.176.15', 'name': 'redis_rpc_perf.default--redis--6791.0', 'job_uniq_id': '6791', 'application': 'redis', 'workload_name': 'redis_rpc_perf', 'env_uniq_id': '15', 'application_version_name': 'default', 'workload_version_name': 'default', 'replica_index': '0', 'application_host_ip': '100.64.176.15', 'load_generator_host_ip': '100.64.176.27', 'load_generator': 'rpc_perf', 'task_id': 'root-staging15-redis_rpc_perf-default--redis--6791-0-0-53b13ed6-c73a-4502-add5-4175ab23ed0b', 'initial_task_cpu_assignment': '1.0'}, 'root-staging15-redis_rpc_perf-default--redis--6789-0-0-90aa8f45-1aba-4024-8854-228bb553e084': {'aurora_tier': 'preemptible', 'job_uniq_id': '6789', 'job_name': 'redis_rpc_perf.default--redis--6789.0', 'own_ip': '100.64.176.15', 'application': 'redis', 'workload_name': 'redis_rpc_perf', 'workload_instance': 'redis_rpc_perf--default--15--6789', 'env_uniq_id': '15', 'name': 'redis_rpc_perf.default--redis--6789.0', 'application_version_name': 'default', 'workload_version_name': 'default', 'replica_index': '0', 'application_host_ip': '100.64.176.15', 'load_generator_host_ip': '100.64.176.27', 'load_generator': 'rpc_perf', 'task_id': 'root-staging15-redis_rpc_perf-default--redis--6789-0-0-90aa8f45-1aba-4024-8854-228bb553e084', 'initial_task_cpu_assignment': '1.0'}, 'root-staging15-twemcache_mutilate-small--twemcache--11211-0-0-c5253ea3-b401-4443-b792-af0f970f322f': {'aurora_tier': 'preemptible', 'application': 'twemcache', 'job_uniq_id': '11211', 'own_ip': '100.64.176.15', 'job_name': 'twemcache_mutilate.small--twemcache--11211.0', 'application_version_name': 'small', 'env_uniq_id': '15', 'name': 'twemcache_mutilate.small--twemcache--11211.0', 'load_generator': 'mutilate', 'workload_version_name': 'small', 'workload_name': 'twemcache_mutilate', 'workload_instance': 'twemcache_mutilate--small--15--11211', 'replica_index': '0', 'application_host_ip': '100.64.176.15', 'load_generator_host_ip': '100.64.176.27', 'task_id': 'root-staging15-twemcache_mutilate-small--twemcache--11211-0-0-c5253ea3-b401-4443-b792-af0f970f322f', 'initial_task_cpu_assignment': '2.0'}, 'root-staging15-cassandra_stress-default--cassandra--9142-0-0-2cf6f910-87c0-44aa-b5f0-230cbcea14e9': {'aurora_tier': 'preemptible', 'load_generator': 'cassandra_stress', 'workload_name': 'cassandra_stress', 'own_ip': '100.64.176.15', 'env_uniq_id': '15', 'workload_instance': 'cassandra_stress--default--15--9142', 'name': 'cassandra_stress.default--cassandra--9142.0', 'application': 'cassandra', 'application_version_name': 'default', 'workload_version_name': 'default', 'job_name': 'cassandra_stress.default--cassandra--9142.0', 'job_uniq_id': '9142', 'replica_index': '0', 'type': 'best_efforts', 'application_host_ip': '100.64.176.15', 'load_generator_host_ip': '100.64.176.27', 'task_id': 'root-staging15-cassandra_stress-default--cassandra--9142-0-0-2cf6f910-87c0-44aa-b5f0-230cbcea14e9', 'initial_task_cpu_assignment': '10.0'}, 'root-staging15-twemcache_mutilate-small--twemcache--11212-0-0-07cdf61b-6832-4639-ae97-a81b65260cd7': {'aurora_tier': 'preemptible', 'application': 'twemcache', 'job_uniq_id': '11212', 'own_ip': '100.64.176.15', 'application_version_name': 'small', 'env_uniq_id': '15', 'load_generator': 'mutilate', 'workload_version_name': 'small', 'workload_name': 'twemcache_mutilate', 'name': 'twemcache_mutilate.small--twemcache--11212.0', 'job_name': 'twemcache_mutilate.small--twemcache--11212.0', 'workload_instance': 'twemcache_mutilate--small--15--11212', 'replica_index': '0', 'application_host_ip': '100.64.176.15', 'load_generator_host_ip': '100.64.176.27', 'task_id': 'root-staging15-twemcache_mutilate-small--twemcache--11212-0-0-07cdf61b-6832-4639-ae97-a81b65260cd7', 'initial_task_cpu_assignment': '2.0'}, 'root-staging15-redis_rpc_perf-default--redis--6790-0-0-69f60671-18d4-488a-ad2e-740cd0620262': {'aurora_tier': 'preemptible', 'job_name': 'redis_rpc_perf.default--redis--6790.0', 'workload_instance': 'redis_rpc_perf--default--15--6790', 'name': 'redis_rpc_perf.default--redis--6790.0', 'own_ip': '100.64.176.15', 'application': 'redis', 'workload_name': 'redis_rpc_perf', 'job_uniq_id': '6790', 'env_uniq_id': '15', 'application_version_name': 'default', 'workload_version_name': 'default', 'replica_index': '0', 'application_host_ip': '100.64.176.15', 'load_generator_host_ip': '100.64.176.27', 'load_generator': 'rpc_perf', 'task_id': 'root-staging15-redis_rpc_perf-default--redis--6790-0-0-69f60671-18d4-488a-ad2e-740cd0620262', 'initial_task_cpu_assignment': '1.0'}, 'root-staging15-twemcache_mutilate-big--twemcache--11213-0-0-a9a7244a-ef48-4238-a310-d5313d1957ba': {'aurora_tier': 'preemptible', 'application': 'twemcache', 'name': 'twemcache_mutilate.big--twemcache--11213.0', 'workload_instance': 'twemcache_mutilate--big--15--11213', 'job_uniq_id': '11213', 'workload_version_name': 'big', 'own_ip': '100.64.176.15', 'application_version_name': 'big', 'env_uniq_id': '15', 'load_generator': 'mutilate', 'workload_name': 'twemcache_mutilate', 'application_host_ip': '100.64.176.15', 'replica_index': '0', 'load_generator_host_ip': '100.64.176.27', 'job_name': 'twemcache_mutilate.big--twemcache--11213.0', 'task_id': 'root-staging15-twemcache_mutilate-big--twemcache--11213-0-0-a9a7244a-ef48-4238-a310-d5313d1957ba', 'initial_task_cpu_assignment': '4.0'}}
2019-06-06 02:41:03,648 DEBUG    {MainThread} [prm.allocator] current tasks_allocations={'root-staging15-redis_rpc_perf-default--redis--6791-0-0-53b13ed6-c73a-4502-add5-4175ab23ed0b': {'cpu_quota': 0.014204545454545454, 'cpu_shares': 1.0, 'rdt': RDTAllocation(name='', l3='L3:0=fffff;1=fffff', mb=None)}, 'root-staging15-redis_rpc_perf-default--redis--6789-0-0-90aa8f45-1aba-4024-8854-228bb553e084': {'cpu_quota': 0.014204545454545454, 'cpu_shares': 1.0, 'rdt': RDTAllocation(name='', l3='L3:0=fffff;1=fffff', mb=None)}, 'root-staging15-twemcache_mutilate-small--twemcache--11211-0-0-c5253ea3-b401-4443-b792-af0f970f322f': {'cpu_quota': 0.02556818181818182, 'cpu_shares': 1.0, 'rdt': RDTAllocation(name='', l3='L3:0=fffff;1=fffff', mb=None)}, 'root-staging15-cassandra_stress-default--cassandra--9142-0-0-2cf6f910-87c0-44aa-b5f0-230cbcea14e9': {'cpu_quota': 0.011363636363636364, 'cpu_shares': 0.002, 'rdt': RDTAllocation(name='BE_Group', l3='L3:0=c0000;1=c0000', mb=None)}, 'root-staging15-twemcache_mutilate-small--twemcache--11212-0-0-07cdf61b-6832-4639-ae97-a81b65260cd7': {'cpu_quota': 0.02556818181818182, 'cpu_shares': 1.0, 'rdt': RDTAllocation(name='', l3='L3:0=fffff;1=fffff', mb=None)}, 'root-staging15-redis_rpc_perf-default--redis--6790-0-0-69f60671-18d4-488a-ad2e-740cd0620262': {'cpu_quota': 0.014204545454545454, 'cpu_shares': 1.0, 'rdt': RDTAllocation(name='', l3='L3:0=fffff;1=fffff', mb=None)}, 'root-staging15-twemcache_mutilate-big--twemcache--11213-0-0-a9a7244a-ef48-4238-a310-d5313d1957ba': {'cpu_quota': 0.048295454545454544, 'cpu_shares': 1.0, 'rdt': RDTAllocation(name='', l3='L3:0=fffff;1=fffff', mb=None)}}
2019-06-06 02:41:03,649 DEBUG    {MainThread} [prm.allocator] metrics: [Metric(name='threshold_lcutilmax', value=1157.2137940532957, labels={}, type=None, help=None), Metric(name=<Metric.LCCAPACITY: 'latency_critical_utilization_capacity'>, value=1100.0, labels={}, type=None, help=None), Metric(name=<Metric.LCMAX: 'latency_critical_utilization_max'>, value=1157.2137940532957, labels={}, type=None, help=None), Metric(name=<Metric.SYSUTIL: 'system_utilization'>, value=751.8525879393407, labels={}, type=None, help=None)]
2019-06-06 02:41:03,650 WARNING  {MainThread} [wca.storage] failed to convert metrics intoprometheus exposition format; error: "Used wrong label name org.apache.aurora.tier in metric allocation_cpu_quota." - skipping
2019-06-06 02:41:04,247 WARNING  {MainThread} [wca.security] Exception <class 'FileNotFoundError'> with message [Errno 2] No such file or directory thrown
2019-06-06 02:41:04,247 ERROR    {MainThread} [wca.resctrl] Could not write pid to resctrl ('/sys/fs/resctrl/mon_groups/mesos-24324f18-3410-4f62-add8-659ded0cba16/tasks'): rdt group was not found (moved/deleted - race detected).
2019-06-06 02:41:04,247 WARNING  {MainThread} [wca.resctrl] Could not close resctrl/tasks file - ignored!(side-effect of previous warning!)
Traceback (most recent call last):
  File ".bootstrap/pex/pex.py", line 352, in execute
  File ".bootstrap/pex/pex.py", line 284, in _wrap_coverage
  File ".bootstrap/pex/pex.py", line 315, in _wrap_profiling
  File ".bootstrap/pex/pex.py", line 397, in _execute
  File ".bootstrap/pex/pex.py", line 495, in execute_entry
  File ".bootstrap/pex/pex.py", line 510, in execute_pkg_resources
  File "/root/.pex/install/wca-0.4.dev37+ge3619f6-py3-none-any.whl.2d043b66053d1f7c89819d8bbac89e56fc623a5e/wca-0.4.dev37+ge3619f6-py3-none-any.whl/wca/main.py", line 127, in main
    exit_code = runner.run()
  File "/root/.pex/install/wca-0.4.dev37+ge3619f6-py3-none-any.whl.2d043b66053d1f7c89819d8bbac89e56fc623a5e/wca-0.4.dev37+ge3619f6-py3-none-any.whl/wca/runners/measurement.py", line 195, in run
    self._iterate()
  File "/root/.pex/install/wca-0.4.dev37+ge3619f6-py3-none-any.whl.2d043b66053d1f7c89819d8bbac89e56fc623a5e/wca-0.4.dev37+ge3619f6-py3-none-any.whl/wca/runners/measurement.py", line 167, in _iterate
    tasks_measurements, tasks_resources, tasks_labels = _prepare_tasks_data(containers)
  File "/root/.pex/install/wca-0.4.dev37+ge3619f6-py3-none-any.whl.2d043b66053d1f7c89819d8bbac89e56fc623a5e/wca-0.4.dev37+ge3619f6-py3-none-any.whl/wca/profiling.py", line 59, in _inner
    result = function_to_profile(*args, **kwargs)
  File "/root/.pex/install/wca-0.4.dev37+ge3619f6-py3-none-any.whl.2d043b66053d1f7c89819d8bbac89e56fc623a5e/wca-0.4.dev37+ge3619f6-py3-none-any.whl/wca/logger.py", line 141, in __trace
    rv = func(*args, **kw)
  File "/root/.pex/install/wca-0.4.dev37+ge3619f6-py3-none-any.whl.2d043b66053d1f7c89819d8bbac89e56fc623a5e/wca-0.4.dev37+ge3619f6-py3-none-any.whl/wca/runners/measurement.py", line 229, in _prepare_tasks_data
    task_measurements = container.get_measurements()
  File "/root/.pex/install/wca-0.4.dev37+ge3619f6-py3-none-any.whl.2d043b66053d1f7c89819d8bbac89e56fc623a5e/wca-0.4.dev37+ge3619f6-py3-none-any.whl/wca/containers.py", line 290, in get_measurements
    self._rdt_information.rdt_cache_monitoring_enabled)
  File "/root/.pex/install/wca-0.4.dev37+ge3619f6-py3-none-any.whl.2d043b66053d1f7c89819d8bbac89e56fc623a5e/wca-0.4.dev37+ge3619f6-py3-none-any.whl/wca/resctrl.py", line 203, in get_measurements
    MON_GROUPS, mongroup_name, MON_DATA)):
FileNotFoundError: [Errno 2] No such file or directory: '/sys/fs/resctrl/mon_groups/mesos-24324f18-3410-4f62-add8-659ded0cba16/mon_data'
```

**Comment by ppalucki** [comment]
it is fixed, @thuang6 can we close that ?

**Comment by thuang6** [comment]
I cannot verify this since I was blocked by another issue 
https://github.com/intel/workload-collocation-agent/issues/154
